lib/events/s3sessions/s3handler.go

```diff
@@ -140,7 +140,7 @@ func NewHandler(cfg Config) (*Handler, error) {
 	}
 	start := time.Now()
 	h.Infof("Setting up bucket %q, sessions path %q in region %q.", h.Bucket, h.Path, h.Region)
-	if err := h.ensureBucket(); err != nil {
+	if err := h.ensureBucket(context.Background()); err != nil {
```

Comment:
This `NewHandler` function should also have a `context.Context` argument. Apparently there is only a single active pathway calling this code so the context would probably originate from https://github.com/gravitational/teleport/blob/994d2c7022237609d02130fae16589ce0a1570a6/lib/service/service.go#L1104-L1105 by passing in `process.ExitContext()`.

If this was too cumbersome, it'd be preferable to use `context.TODO()` instead of `context.Background()` here.

Reply:
I used `context.Background` to mimic the [gcs](https://github.com/gravitational/teleport/blob/994d2c7022237609d02130fae16589ce0a1570a6/lib/events/gcssessions/gcshandler.go#L178) behavior. Now using your suggestion and piping the `process.ExitContext()` into both of them.